arangod/RestHandler/RestVocbaseBaseHandler.cpp

```diff
@@ -487,7 +488,7 @@ void RestVocbaseBaseHandler::generateTransactionError(std::string const& collect
       return;
 
     case TRI_ERROR_ARANGO_CONFLICT:
-      if (result.buffer != nullptr) {
+      if (result.buffer != nullptr && !result.slice().isNone()) {
         // This case happens if we come via the generateTransactionError that
```

Comment:
Is this consistent with the behavior in the other crud methods now ?

Reply:
I don't know. But if the slice is `None`, we will not be able to serialize it to JSON later on.